service/history/api/consistency_checker.go

```diff
@@ -178,9 +179,91 @@ func (c *WorkflowConsistencyCheckerImpl) getWorkflowLeaseValidatedByClock(
 		release(err)
 		return nil, err
 	}
+
 	return NewWorkflowLease(wfContext, release, mutableState), nil
 }
 
+func (c *WorkflowConsistencyCheckerImpl) getCurrentWorkflowContext(
+	ctx context.Context,
+	namespaceID string,
+	workflowID string,
+	lockPriority workflow.LockPriority,
+) (WorkflowLease, error) {
+	runID, err := c.GetCurrentRunID(
+		ctx,
+		namespaceID,
+		workflowID,
+		lockPriority,
+	)
+	if err != nil {
+		return nil, err
+	}
+	workflowLease, err := c.getWorkflowLease(
+		ctx, definition.NewWorkflowKey(namespaceID, workflowID, runID), lockPriority,
+	)
+	if err != nil {
+		return nil, err
+	}
+	if workflowLease.GetMutableState().IsWorkflowExecutionRunning() {
+		return workflowLease, nil
+	}
+
+	currentRunID, err := c.GetCurrentRunID(ctx, namespaceID, workflowID, lockPriority)
+
+	if err != nil {
+		workflowLease.GetReleaseFn()(err)
+		return nil, err
+	}
+	if currentRunID == workflowLease.GetContext().GetWorkflowKey().RunID {
+		return workflowLease, nil
+	}
+
+	workflowLease.GetReleaseFn()(nil)
+	return nil, consts.ErrLocateCurrentWorkflowExecution
+}
+
+// The code below is an old implementation of GetWorkflowLease
+// It should be avoided. We should revisit the logic
```

Comment:
Plz update the comment. Consistency is necessary when shard clock is not applicable and request can't operate on stale mutable state.

Reply:
done